src/base/http/server.cpp

```diff
@@ -70,20 +82,34 @@ void Server::incomingConnection(qintptr socketDescriptor)
 #endif
         serverSocket = new QTcpSocket(this);
 
-    if (serverSocket->setSocketDescriptor(socketDescriptor)) {
+    if (!serverSocket->setSocketDescriptor(socketDescriptor)) {
+        delete serverSocket;
+        return;
+    }
+
 #ifndef QT_NO_OPENSSL
-        if (m_https) {
-            static_cast<QSslSocket *>(serverSocket)->setProtocol(QSsl::SecureProtocols);
-            static_cast<QSslSocket *>(serverSocket)->setPrivateKey(m_key);
-            static_cast<QSslSocket *>(serverSocket)->setLocalCertificateChain(m_certificates);
-            static_cast<QSslSocket *>(serverSocket)->setPeerVerifyMode(QSslSocket::VerifyNone);
-            static_cast<QSslSocket *>(serverSocket)->startServerEncryption();
-        }
-#endif
-        new Connection(serverSocket, m_requestHandler, this);
+    if (m_https) {
+        static_cast<QSslSocket *>(serverSocket)->setProtocol(QSsl::SecureProtocols);
+        static_cast<QSslSocket *>(serverSocket)->setPrivateKey(m_key);
+        static_cast<QSslSocket *>(serverSocket)->setLocalCertificateChain(m_certificates);
+        static_cast<QSslSocket *>(serverSocket)->setPeerVerifyMode(QSslSocket::VerifyNone);
+        static_cast<QSslSocket *>(serverSocket)->startServerEncryption();
     }
-    else {
-        serverSocket->deleteLater();
+#endif
+
+    Connection *c = new Connection(serverSocket, m_requestHandler, this);
+    m_connections.append(c);
+}
+
+void Server::dropTimeoutConnections()
```

Comment:
The `Timeout` word is actually used in the specs:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Keep-Alive

Since "timeout + ed" is not a word, how about `dropTimedoutConnections()`?
got the idea from: https://github.com/jbosstools/jbosstools-openshift/pull/144/files